tests/test_database_matrix.py

```diff
@@ -0,0 +1,1677 @@
+        opsdroid=opsdroid_matrix,
+    )
+    db.should_migrate = False
+    await db.put("twim", {"hello": "world"})
+
+    patched_send.assert_has_calls(
+        [
+            matrix_call(
+                "GET",
+                "/_matrix/client/r0/rooms/%21notaroomid/state/dev.opsdroid.database/wibble",
+            ),
+            matrix_call(
+                "PUT",
+                "/_matrix/client/r0/rooms/%21notaroomid/state/dev.opsdroid.database/wibble",
+                {"twim": {"hello": "world"}},
+            ),
+        ],
+    )
+
+
+@pytest.mark.asyncio
+@pytest.mark.xfail(
+    not nio.crypto.ENCRYPTION_ENABLED, reason="No encryption deps installed for matrix"
+)
+async def test_put_custom_state_key_enc(patched_send, opsdroid_matrix, patched_uuid):
+    def side_effect(resp, *args, **kwargs):
+        if resp is nio.RoomGetStateEventResponse:
+            resp = nio.RoomGetStateEventResponse({}, "", "", "")
+            resp.transport_response = AsyncMock()
+            resp.transport_response.status = 404
+            return resp
+        else:
+            return nio.RoomSendResponse("enceventid", "!notaroomid")
+
+    patched_send.side_effect = side_effect
+
+    db = DatabaseMatrix({"single_state_key": "wibble"}, opsdroid=opsdroid_matrix)
+    db.should_migrate = False
+    await db.put("twim", {"hello": "world"})
+
+    patched_send.assert_has_calls(
+        [
+            matrix_call(
+                "GET",
+                "/_matrix/client/r0/rooms/%21notaroomid/state/dev.opsdroid.database/wibble",
+            ),
+            matrix_call(
+                "PUT",
+                "/_matrix/client/r0/rooms/%21notaroomid/send/dev.opsdroid.database/",
+                {"twim": {"hello": "world"}},
+            ),
+            matrix_call(
+                "PUT",
+                "/_matrix/client/r0/rooms/%21notaroomid/state/dev.opsdroid.database/wibble",
+                {"twim": {"encrypted_val": "enceventid"}},
+            ),
+        ],
+        any_order=True,
+    )
+
+
+@pytest.mark.asyncio
+async def test_single_state_key_false(patched_send, opsdroid_matrix):
+    patched_send.return_value = nio.RoomGetStateEventResponse({}, "", "", "")
+    patched_send.return_value.transport_response = AsyncMock()
+    patched_send.return_value.transport_response.status = 404
+
+    db = DatabaseMatrix(
+        {"should_encrypt": False, "single_state_key": False}, opsdroid=opsdroid_matrix
+    )
+    db.should_migrate = False
+    await db.put("hello", "world")
+
+    patched_send.assert_has_calls(
+        [
+            matrix_call(
+                "GET",
+                "/_matrix/client/r0/rooms/%21notaroomid/state/dev.opsdroid.database/hello",
+            ),
+            matrix_call(
+                "PUT",
+                "/_matrix/client/r0/rooms/%21notaroomid/state/dev.opsdroid.database/hello",
+                {"hello": "world"},
+            ),
+        ],
+    )
+
+
+@pytest.mark.asyncio
+@pytest.mark.xfail(
```

Comment:
I think skip might be better than `xfail` if the deps aren't installed?

Reply:
We're mocking the methods that encryption would affect anyway so without deps they would still fail correctly. Plus in that case we won't know if they xpass after future changes.